src/MongoDB.Analyzer/Core/Utilities/SyntaxNodeExtensions.cs

```diff
@@ -34,7 +34,7 @@ internal static class SyntaxNodeExtensions
     public static MethodDeclarationSyntax GetSingleMethod(this SyntaxNode syntaxNode, string name) =>
         syntaxNode.DescendantNodes().OfType<MethodDeclarationSyntax>().Single(m => m.Identifier.Text == name);
 
-    public static SyntaxNode GetTopMostInvocationOrBinaryExpressionSyntax(SimpleNameSyntax identifier, IdentifierNameSyntax[] lambdaIdentifiers)
+    public static SyntaxNode GetTopMostInvocationOrBinaryExpressionSyntax(SimpleNameSyntax identifier, IdentifierNameSyntax[] lambdaIdentifiers, IdentifierNameSyntax [] queryIdentifiers = null)
```

Comment:
Do we really need to distinguish between lambdaIdentifiers and queryIdentifiers ?